test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -5477,6 +5483,54 @@ BOOST_AUTO_TEST_CASE(using_interface_complex)
 	success(text);
 }
 
+BOOST_AUTO_TEST_CASE(warn_unused_local)
+{
+	char const* text = R"(
+		contract C {
+			function f() {
+				uint a;
+			}
+		}
+  )";
+	CHECK_WARNING(text, "Unused");
+}
+
+BOOST_AUTO_TEST_CASE(warn_unused_param)
+{
+	char const* text = R"(
+		contract C {
+			function f(uint a) {
```

Comment:
It can be useful if you have either default implementations or overriden methods that don't make use of all parameters in the declared interface.